src/components/TextField/TextField.jsx

```diff
@@ -0,0 +1,75 @@
+import { TextField as MUITextField } from '@mui/material';
+import InputAdornment from '@mui/material/InputAdornment';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import textFieldStyles from './TextFieldStyles';
+
+const TextField = (props) => {
+    const {
+        placeholder,
+        error,
+        position,
+        children,
+        adornment,
+        disabled,
+        helperText,
+        label,
+        type,
+        className,
+        ...rest
+    } = props;
+
+    const [isFocused, setIsFocused] = useState(false);
+
+    const onFocusHandler = () => {
+        setIsFocused(true);
+    };
+    const onBlurHandler = () => {
+        setIsFocused(false);
+    };
+
+    return (
+        <MUITextField
+            sx={{
+                width: 260,
+                minHeight: 56,
+                ...textFieldStyles[type || 'default'],
+            }}
+            placeholder={placeholder}
+            label={isFocused ? label : null}
+            error={error}
+            disabled={disabled}
+            helperText={disabled ? null : error ? helperText : null}
```

Comment:
You should go away from hard, complex logic and avoid double ternar operator here.